src/logservice/logminer/ob_log_miner_logger.cpp

```diff
@@ -60,15 +64,24 @@ int LogMinerLogger::log_progress(int64_t record_num, int64_t current_ts, int64_t
     percentage = 1;
   }
   progress = percentage * 100;
-  lpad = (int)(percentage * PB_WIDTH);
-  rpad = PB_WIDTH - lpad;
+
+  ioctl(STDOUT_FILENO, TIOCGWINSZ, &w); // get the width of terminal window
+  // 68: 20->datatime, 2->[], 8->percentage, 18->" written records: ", 20->the length of INT64_MAX
+  pb_width = w.ws_col - 68;
+  if (pb_width < 5) {
+    // the length of terminal window is too short, fixed-width display
+    pb_width = 50;
```

Comment:
`pb_width = 5` is better?

Reply:
I think `pb_width = 50` a little strange when `w.ws_col - 68 =4` and `pb_width = 5` when `w.ws_col - 68 =5`

Reply:
> I think `pb_width = 50` a little strange when `w.ws_col - 68 =4` and `pb_width = 5` when `w.ws_col - 68 =5`

It makes sense, maybe `pb_width = (pb_width < 50) ? 50 : pb_width` is better.

Reply:
> > I think `pb_width = 50` a little strange when `w.ws_col - 68 =4` and `pb_width = 5` when `w.ws_col - 68 =5`
> 
> It makes sense, maybe `pb_width = (pb_width < 50) ? 50 : pb_width` is better.

I think minimal pb_width = 5 is ok. Because there are many people who like to open multiple windows at the same time, resulting in the size of a single window is relatively small.
You can cancel the printing of the progress bar when the command line width is too small. See `wget` for how to handle this.